### ISSUE TITLE
kubectl create help implies a list of service accounts can be references in a clusterrolebinding

### ISSUE THREAD

**Issue by RichardoC** [kubernetes/kubernetes#108608]
### What happened?

I'm not sure whether this is the documentation being unclear, or a bug with kubectl itself but I suspect the former.

Given this help page
```bash
$ kubectl create clusterrolebinding --help
Create a cluster role binding for a particular cluster role.
# Omitted for brevity
      --serviceaccount=[]: Service accounts to bind to the clusterrole, in the format <namespace>:<name>
```
I expected to be able to use
```bash
$ kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
```
Instead I got the following error
```bash
$ kubectl create clusterrolebinding pipelinesas --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
error: exactly one NAME is required, got 2
See 'kubectl create clusterrolebinding -h' for help and examples
```

### What did you expect to happen?

To get the same output as defining the flag twice

aka
```bash
$ kubectl create clusterrolebinding pipelinesas --clusterrole=view --serviceaccount=ns1:pipeline --serviceaccount=ns2:pipeline
clusterrolebinding.rbac.authorization.k8s.io/pipelinesas created
```
and

```bash
$ kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
clusterrolebinding.rbac.authorization.k8s.io/pipelinesas created
```
should give the same output, or the help page shouldn't list `[]` as the parameter type for --serviceaccount

### How can we reproduce it (as minimally and precisely as possible)?

```bash
kubectl create ns ns1 
kubectl create ns ns2
kubectl --namespace ns1 create sa pipeline
kubectl --namespace ns2 create sa pipeline
kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=["ns1:pipeline", "ns2:pipeline"]
```

### Kubernetes version

<details>

```console
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:34:54Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
N/A
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -a
Linux controlplane 5.4.0-88-generic #99-Ubuntu SMP Thu Sep 23 17:29:00 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

# On Windows:
N/A
```

</details>

### Install tools

<details>
```console
$ kubectl -n kube-system get cm kubeadm-config  -oyaml
apiVersion: v1
data:
  ClusterConfiguration: |
    apiServer:
      extraArgs:
        authorization-mode: Node,RBAC
      timeoutForControlPlane: 4m0s
    apiVersion: kubeadm.k8s.io/v1beta3
    certificatesDir: /etc/kubernetes/pki
    clusterName: kubernetes
    controllerManager: {}
    dns: {}
    etcd:
      local:
        dataDir: /var/lib/etcd
    imageRepository: k8s.gcr.io
    kind: ClusterConfiguration
    kubernetesVersion: v1.23.1
    networking:
      dnsDomain: cluster.local
      podSubnet: 192.168.0.0/16
      serviceSubnet: 10.96.0.0/12
    scheduler: {}
kind: ConfigMap
metadata:
  name: kubeadm-config
  namespace: kube-system
```
</details>

### Container runtime (CRI) and version (if applicable)

<details>
      architecture: amd64
      bootID: 09b2e311-c6dc-40bb-9485-64c98f77cf66
      containerRuntimeVersion: containerd://1.5.5
      kernelVersion: 5.4.0-88-generic
      kubeProxyVersion: v1.23.1
      kubeletVersion: v1.23.1
</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by liggitt** [comment]
the flag can be specified multiple times, or (I think) as a comma-separated list. agree the `[]` notation is confusing... I think that's how cobra prints the default value (which is an empty list)

**Comment by Shubham82** [comment]
Hi @liggitt @RichardoC

> the flag can be specified multiple times, or (I think) as a comma-separated list. agree the `[]` notation is confusing... I think that's how cobra prints the default value (which is an empty list)

I have tried and confirmed that it works with multiple flags:

```
$ kubectl create clusterrolebinding pipelineSAs --clusterrole=view --serviceaccount=ns1:pipeline --serviceaccount=ns2:pipeline
   
   clusterrolebinding.rbac.authorization.k8s.io/pipelineSAs created
```
But for the comma-separated list, doesn't work. 

Kubectl version: v1.23.3  and k8s version v1.23.3

**Comment by Shubham82** [comment]
The same update for rolebinding:

```
$ kubectl create  rolebinding --help
Create a role binding for a particular role or cluster role.

      --serviceaccount=[]: Service accounts to bind to the role, in the format <namespace>:<name>
```
Should we write like this(same for clusterrolebinding):
 **`--serviceaccount=[]: Service accounts to bind to the role, in the format <namespace>:<name>. To bind multiple serviceaccount to the role, specify the flag multiple times.`**

**Comment by RichardoC** [comment]
Thanks for the investigation, another approach that might help is adding an example with the comma separated service accounts in the examples section. 

I do wonder if there's a clearer representation that could be output by cobra, to make it clear what the behaviour should be...?

**Comment by Shubham82** [comment]
Hi, @RichardoC  I think adding the example is good so that users who reference this document get the proper information.
but there is already mention of how to this flag with create command under the usage section:

```
kubectl create clusterrolebinding -h
Create a cluster role binding for a particular cluster role.
Usage:
  kubectl create clusterrolebinding NAME --clusterrole=NAME [--user=username] [--group=groupname]
[--serviceaccount=namespace:serviceaccountname] [--dry-run=server|client|none] [options]

```
What do you suggest @liggitt?

**Comment by liggitt** [comment]
in the specific flag help, we can mention the flag can be repeated to add multiple subjects

**Comment by RichardoC** [comment]
> > in the specific flag help, we can mention the flag can be repeated to add multiple subjects
> 
> +1 I will update this. /assign

This sounds good to me, might be worth doing the same for other places that use the same behaviour?

**Comment by Shubham82** [comment]
Sure @RichardoC 
I will do that.

**Comment by Shubham82** [comment]
I have fixed it here: #108654
PTAL!
